irohad/main/application.cpp

```diff
@@ -59,7 +59,8 @@ Irohad::Irohad(const std::string &block_store_dir,
       max_proposal_size_(max_proposal_size),
       proposal_delay_(proposal_delay),
       vote_delay_(vote_delay),
-      is_mst_supported_(is_mst_supported),
+      is_mst_supported_(opt_mst_gossip_params),
```

Comment:
Is that field is still required?

Reply:
Yes, it is used two times. I do not want to substitute it with optional gossip params variable, because an explicit bool is much more expressive.

Reply:
If we add a separate constructor for every optional feature - we will end up with 2^n constructors =). Also it would involve copy-pasting common parts of constructor code. So I do not see the benefits of such approach. Please clarify it if I miss something.

Reply:
> I guess maybe better to add another ctor of irohad, one will not receive both is_mst_supported and params and another will receive only params without optional wrapper.

If we add a separate constructor for every optional feature - we will end up with 2^n constructors =). Also it would involve copy-pasting common parts of constructor code. So I do not see the benefits of such approach. Please clarify it if I miss something.

Reply:
> It is more error-prone to track two variables though, because of the opt_mst_gossip_params => is_mst_supported_ relation.
It is a part of internal logic stated explicitly in the constructor: we enable the MST support if we got parameters for its initialization. After that we use each of the two variables only in its own sense, and I think it is less error prone than if we would check the parameters presence each time we need to know whether we support MST, as that would be  real mess.